From c70b1dc10a8c094fb421f2deaea5cd2fee69e016 Mon Sep 17 00:00:00 2001
From: Pankaj Bansal <pankaj.bansal@nxp.com>
Date: Tue, 27 Nov 2018 00:18:23 +0530
Subject: [PATCH 600/706] can: flexcan: Add warning if prescaler is not uniform

As per can for Auto guidelines the prescaler for data phase and control phase
is desired to be same.
Since this value is calculated by can driver (same for all can controllers),
it is beyond the control of flexcan.
Therefore, just put a warning if the mismatch occurs between two prescaler
values, so that user is able to adjust the bitrate to make the prescaler
uniform.

Signed-off-by: Pankaj Bansal <pankaj.bansal@nxp.com>
(cherry picked from commit 8d0ee098fd8c79e2df5ec5ba140582712e083189)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/can/flexcan.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/can/flexcan.c b/drivers/net/can/flexcan.c
index 7fecd97283d1..b5607b312736 100644
--- a/drivers/net/can/flexcan.c
+++ b/drivers/net/can/flexcan.c
@@ -1112,6 +1112,9 @@ static void flexcan_set_bittiming(struct net_device *dev)
 			reg_fdctrl |= FLEXCAN_FDCTRL_TDCOFF(tdcoff);
 			reg_fdctrl |= FLEXCAN_FDCTRL_MBDSR0(3) |
 				      FLEXCAN_FDCTRL_MBDSR1(3);
+			if (data_bt->brp != bt->brp)
+				netdev_warn(dev, "data brp = %d, brp = %d",
+					    data_bt->brp, bt->brp);
 		} else {
 			reg_fdctrl |= FLEXCAN_FDCTRL_MBDSR0(0) |
 				      FLEXCAN_FDCTRL_MBDSR1(0);
-- 
2.17.1

